pkg/operator/controller/controller_dns.go

```diff
@@ -15,8 +15,12 @@ import (
 
 // ensureDNS will create DNS records for the given LB service. If service is
 // nil, nothing is done.
-func (r *reconciler) ensureDNS(ci *ingressv1alpha1.ClusterIngress, service *corev1.Service, dnsConfig *configv1.DNS) error {
-	dnsRecords, err := desiredDNSRecords(ci, service, dnsConfig)
+func (r *reconciler) ensureDNS(ci *ingressv1alpha1.ClusterIngress, service *corev1.Service, dnsConfig *configv1.DNS, ingressDomain *string, ha *ingressv1alpha1.ClusterIngressHighAvailability) error {
+	if ha == nil {
```

Comment:
Passing a pointer to this seems really weird... we have a None type and should have come up with the effective value before calling this, so can it ever even be nil? What about checking that ha type is Cloud?

Reply:
Just wasn't sure we wanted to pass the struct itself rather than a pointer.  I can drop the `*`.

Reply:
Fixed.  Thanks!